util/mcf_dict_util.py

```diff
@@ -0,0 +1,537 @@
+    print(req.request.url)
+
+    if req.status_code == requests.codes.ok:
+        response_data = req.json()
+    else:
+        response_data = {'http_err_code': req.status_code}
+        print('HTTP status code: ' + str(req.status_code))
+    return response_data
+
+
+def dc_check_existence(dcid_list: list,
+                       use_autopush: bool = True,
+                       max_items: int = 450) -> dict:
+    """Checks if a given list of dcids are present in DC.
+
+    Args:
+        dcid_list: List of dcids to be queried for existence.
+        use_autopush: Boolean value to use autopush API and not public API.
+        max_items: Limit of items to be queried in a single POST request.
+
+
+    Returns:
+        Dict object with dcids as key values and boolean values signifying existence as values.
+    """
+    data_ = {}
+    ret_dict = {}
+    if use_autopush:
+        url_prefix = 'autopush.'
+    else:
+        url_prefix = ''
+
+    chunk_size = max_items
+    dcid_list_chunked = [
+        dcid_list[i:i + chunk_size]
+        for i in range(0, len(dcid_list), chunk_size)
+    ]
+    for dcid_chunk in dcid_list_chunked:
+        data_["dcids"] = dcid_chunk
+        req = request_post_json(
+            f'https://{url_prefix}api.datacommons.org/node/property-labels',
+            data_)
+        resp_dicts = req['payload']
+        resp_dicts = ast.literal_eval(resp_dicts)
+        for cur_dcid in resp_dicts:
+            if not resp_dicts[cur_dcid]:
+                ret_dict[cur_dcid] = False
+            elif not resp_dicts[cur_dcid]['inLabels'] and not resp_dicts[
+                    cur_dcid]['outLabels']:
+                ret_dict[cur_dcid] = False
+            else:
+                ret_dict[cur_dcid] = True
+
+    return ret_dict
+
+
+def mcf_to_dict_list(mcf_str: str) -> list:
+    """Convets MCF file string to a list of OrderedDict objects.
+
+    Args:
+        mcf_str: String read from MCF file.
+        
+    Returns:
+        List of OrderedDict objects where each object represents a node in the MCF file.
+    """
+    # TODO preserve empty lines if required
+    # split by \n\n
+    nodes_str_list = mcf_str.split('\n\n')
+    # each node
+    node_list = []
+
+    for node in nodes_str_list:
+        node = node.strip()
+        # check for comments
+        node_str_list = node.split('\n')
+        cur_node = OrderedDict()
+        comment_ctr = 0
+        is_first_prop = True
+        # add each pv to ordered dict
+        for pv_str in node_str_list:
+            # TODO handle multiple occurrences of same property within a node
+            pv_str = pv_str.strip()
+            if pv_str.startswith('#'):
+                cur_node[f'__comment{comment_ctr}'] = pv_str
+                comment_ctr += 1
+            elif is_first_prop:
+                is_first_prop = False
+                if pv_str and not pv_str.startswith('Node: '):
+                    raise ValueError('Each node must start with Node: <name>".')
```

Comment:
Pls add the line content with the error as well, say,
`raise ValueError('Each node must start with Node: <name>" for node {node}')`

Reply:
fixed.